src/main/java/com/bobocode/petros/container/ApplicationAnnotationContainer.java

```diff
@@ -17,12 +18,30 @@ public ApplicationAnnotationContainer(String packageName) {
 
     @Override
     public <T> T getDependency(String name, Class<T> clazz) {
-
-        return null;
+        return (T) dependencyMap.values().stream()
+                .filter(obj -> obj.getClass().getName().toLowerCase().equals(name.toLowerCase())
+                        && obj.getClass().equals(clazz))
+                .findFirst();
```

Comment:
Method findFirst() is returning Optional, so you have to add orElse(null) or something...